pkg/reconcile/reconcilers.go

```diff
@@ -286,6 +286,20 @@ func (r *defaultReconciler) ReconcileDeployment(ctx context.Context, req ReaperR
 			return &ctrl.Result{RequeueAfter: 10 * time.Second}, err
 		}
 	} else {
+		if deployment.Spec.Strategy.Type != appsv1.RecreateDeploymentStrategyType {
+			// When strategy is not set to RecreateDeploymentStrategyType this can cause [issues](https://github.com/k8ssandra/reaper-operator/issues/63) where multiple Reaper pods exist concurrently.
+			// This is not supported due to the way that migrations in Reaper work. We need to ensure that RecreateDeploymentStrategyType is in place before any upgrades can occur.
+			req.Logger.Info("Reaper deployment not currently using appsv1.RecreateDeploymentStrategyType, patching and re-queueing the request")
+			patch := client.MergeFrom(deployment.DeepCopy())
+			deployment.Spec.Strategy = appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			}
+			if err := r.Patch(ctx, deployment, patch); err != nil {
```

Comment:
As noted in another comment, this .Patch command is probably not going to work due to the wrong patching strategy.

Reply:
It "works" but it does seem to lead to a new replicaset being created in some instances (hard to observe). 

Can you provide me any docs on the `retainKeys` strategy? I have come across [mention](https://stackoverflow.com/questions/67791967/changing-the-deployment-strategy-from-rollingupdate-to-recreate-using-golang) of this before, do I need to use the `dynamic` client or something of that nature?

Reply:
controller-runtime's client should work, I guess the patch strategy of StrategicMergePatchType should do the trick (I'm assuming it defaults to SSA).

Reply:
OK I'll investigate. It may be best to write the integration test first, prove it is failing under the current use of `client-go`, and then prove it passes when using `controller-runtime` client. Feedback on that approach most welcome.

Reply:
This code isn't using client-go directly, it is using controller-runtime client. But the Patch strategy should change.

Reply:
Updating to a newer `controller-runtime` version enabled me to fix this (I think). Resolving.

Reply:
I think this is fixed after updating controller-runtime and switching to `client.StrategicMergeFrom()`.